pykern/fconf.py

```diff
@@ -0,0 +1,517 @@
+# -*- coding: utf-8 -*-
+"""Python and YAML configuration file parser.
+
+FConf reads Python and YAML files and produces a single, merged
+PKDict. FConf is not a replacement for `pykern.pkconfig`. Rather it is
+for complex configuration input files to programs that often require
+programmatic generation. FConf was written for
+RSConf <https://git.radiasoft.org/rsconf>.
+
+The Basic YAML configuration look like this::
+
+    fconf_macros:
+      make_uri(host):
+          "https://${host}"
+
+    some:
+       - item
+    uri: make_uri('sirepo.com')
+
+Macros in the `fconf_macros` are simple replacements.  They cannot have
+loops or other conditions. These are reserved for Python files.
+
+A macro use must be a full text element. It cannot be embedded in
+text, e.g.  ``curl make_uri('sirepo.com')``, that's because macros can
+return data structures. The syntax is pure Python. The arguments have
+to quoted.
```

Comment:
This conflicts with the integer statement below which I found confusing. Are yaml macro args the only ones that have to be quoted?